src/freenet/node/NodeStats.java

```diff
@@ -473,10 +473,16 @@ public void set(Long val) throws InvalidConfigValueException, NodeNeedRestartExc
 
 			@Override
 			public void onChange(NETWORK_THREAT_LEVEL oldLevel, NETWORK_THREAT_LEVEL newLevel) {
-				if(newLevel == NETWORK_THREAT_LEVEL.MAXIMUM)
-					ignoreLocalVsRemoteBandwidthLiability = true;
-				if(oldLevel == NETWORK_THREAT_LEVEL.MAXIMUM)
-					ignoreLocalVsRemoteBandwidthLiability = false;
+				if(newLevel == NETWORK_THREAT_LEVEL.MAXIMUM) {
+					synchronized(NodeStats.this) {
+						ignoreLocalVsRemoteBandwidthLiability = true;
+					}
+				}
+				if(oldLevel == NETWORK_THREAT_LEVEL.MAXIMUM) {
+					synchronized(NodeStats.this) {
+						ignoreLocalVsRemoteBandwidthLiability = false;
+					}
+				}
```

Comment:
While correct as long as `oldLevel != MAXIMUM || newLevel != MAXIMUM`, it feels weird to first update based on the new value, and then on the old value.